src/psyclone/tests/core/signature_test.py

```diff
@@ -173,3 +178,64 @@ def test_signature_comparison():
         _ = Signature(("a", "b")) >= (1, 2)
     assert "'>=' not supported between instances of 'Signature' and 'tuple'" \
         in str(err.value)
+
+
+def test_to_fortran():
+    '''Test that conversion of a Signature with a ComponentIndices argument
+    gives the expected results.
+    '''
+    sig = Signature("a")
+    comp = ComponentIndices()
+    assert sig.to_language(comp) == "a"
+
+    comp = ComponentIndices(["i"])
+    assert sig.to_language(comp) == "a(i)"
+
+    ref = Reference(DataSymbol("j", INTEGER_SINGLE_TYPE))
+    comp = ComponentIndices([ref])
+    assert sig.to_language(comp) == "a(j)"
+
+    # Test error condition if number of components in signature does not
+    # math the number of indices in ComponentIndices
+    comp = ComponentIndices([[1], [2]])
+    with pytest.raises(InternalError) as err:
+        sig.to_language(comp)
+    assert "Signature a has 1 components, but component_indices [[1], [2]] " \
```

Comment:
Quotes around the signature, i.e. "Signature 'a' has ...."

Reply:
Done, and I noticed one or two more messages without '' - not saying I fixed them all, but it's better now :)

Reply:
Thanks Joerg.